src/handlers/issue/generate-nft-mint-signature.ts

```diff
@@ -0,0 +1,127 @@
+import { BigNumber, BigNumberish, ethers, utils } from "ethers";
+import { getPayoutConfigByNetworkId } from "../../helpers/payout";
+import { MaxUint256 } from "@uniswap/permit2-sdk";
+import { keccak256, toUtf8Bytes } from "ethers/lib/utils";
+
+const NFT_MINTER_PRIVATE_KEY = process.env.NFT_MINTER_PRIVATE_KEY as string;
+const NFT_CONTRACT_ADDRESS = process.env.NFT_CONTRACT_ADDRESS as string;
+const NFT_CONTRACT_CHAIN_ID = parseInt(process.env.NFT_CONTRACT_CHAIN_ID as string);
+const SIGNING_DOMAIN_NAME = "NftReward-Domain";
+const SIGNING_DOMAIN_VERSION = "1";
```

Comment:
What is this for?

Reply:
this is used for signing the NFT mint request

Reply:
`SIGNING_DOMAIN_VERSION`

Reply:
Seems odd to me that we have to pass this in as a parameter. @rndquu rfc

Reply:
> User
is `string version` mandatory to pass in as a param

> ChatGPT
Yes, in the context of EIP-712, the string version parameter is mandatory. This is because EIP-712 specifies a structured data hashing and signing mechanism that relies on a versioning system. The version parameter, typically a string, helps in identifying the specific version of the EIP-712 standard being used. This is important for ensuring compatibility and understanding the structure of the data being signed, as different versions might have slight variations in their implementation.

Reply:
I'm pretty sure if we change any of the fields or skip them, the hash will be different and NFT mint will fail, but @rndquu can tell for sure

Reply:
> I'm pretty sure if we change any of the fields or skip them, the hash will be different and NFT mint will fail, but @rndquu can tell for sure

Exactly, contract version is required, pls check [this](https://github.com/ubiquibot/comment-incentives/pull/4#discussion_r1470136916) comment